File: services/management-service/domain/service/system_setting.go
package service

import (
	"context"

	"jusha/gantt/service/management/domain/model"
)

// ISystemSettingService 系统设置领域服务接口
type ISystemSettingService interface {
	// GetSetting 获取系统设置（如果不存在返回默认值）
	GetSetting(ctx context.Context, orgID, key string) (string, error)

	// SetSetting 设置系统设置
	SetSetting(ctx context.Context, orgID, key, value, description string) error

	// GetAllSettings 获取组织的所有设置
	GetAllSettings(ctx context.Context, orgID string) ([]*model.SystemSetting, error)

	// DeleteSetting 删除系统设置
	DeleteSetting(ctx context.Context, orgID, key string) error
}

File: frontend/web/src/api/systemSetting.ts
import { request } from '@/utils/request'

export interface SystemSetting {
  id: string
  orgId: string
  key: string
  value: string
  description?: string
  createdAt: string
  updatedAt: string
}

export interface SystemSettingResponse {
  key: string
  value: string
}

/**
 * 获取所有系统设置
 */
export function getAllSettings(orgId: string) {
  return request<SystemSetting[]>({
    url: '/v1/system-settings',
    method: 'get',
    params: { orgId },
  })
}

/**
 * 获取单个系统设置
 */
export function getSetting(orgId: string, key: string) {
  return request<SystemSettingResponse>({
    url: `/v1/system-settings/${key}`,
    method: 'get',
    params: { orgId },
  })
}

/**
 * 设置系统设置
 */
export function setSetting(orgId: string, key: string, value: string, description?: string) {
  return request<SystemSettingResponse>({
    url: `/v1/system-settings/${key}`,
    method: 'put',
    params: { orgId },
    data: {
      value,
      description,
    },
  })
}

/**
 * 删除系统设置
 */
export function deleteSetting(orgId: string, key: string) {
  return request({
    url: `/v1/system-settings/${key}`,
    method: 'delete',
    params: { orgId },
  })
}
